### PR TITLE
[flutter_tools] require cmdline-tools for android licenses

### LINKED ISSUE #80273: Provide helpful error message when `flutter doctor --android-licenses` fails
## Steps to Reproduce

1. flutter doctor --android-licenses

## System info

```
Host Name:                 MLW223
OS Name:                   Майкрософт Windows 10 Корпоративная LTSC
OS Version:                10.0.17763 N/A Build 17763
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Member Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          lcsa
Registered Organization:
Product ID:                00425-00000-00002-AA147
Original Install Date:     23.09.2020, 13:18:22
System Boot Time:          12.04.2021, 8:32:18
System Manufacturer:       Default string
System Model:              Default string
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 158 Stepping 9 GenuineIntel ~2904 Mhz
BIOS Version:              American Megatrends Inc. 5.12, 16.02.2017
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             ru;Russian
Input Locale:              ru;Russian
Time Zone:                 (UTC+03:00) Minsk
Total Physical Memory:     8 098 MB
Available Physical Memory: 3 219 MB
Virtual Memory: Max Size:  16 802 MB
Virtual Memory: Available: 10 510 MB
Virtual Memory: In Use:    6 292 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    motorland.loc
Logon Server:              \\PDC01
Hotfix(s):                 16 Hotfix(s) Installed.
                           [01]: KB4601055
                           [02]: KB4465065
                           [03]: KB4470788
                           [04]: KB4486153
                           [05]: KB4486174
                           [06]: KB4487038
                           [07]: KB4535680
                           [08]: KB4570332
                           [09]: KB4577586
                           [10]: KB4577667
                           [11]: KB4580325
                           [12]: KB4587735
                           [13]: KB4589208
                           [14]: KB4601393
                           [15]: KB5000859
                           [16]: KB5000822
Network Card(s):           2 NIC(s) Installed.
                           [01]: Intel(R) Dual Band Wireless-AC 3160
                                 Connection Name: Беспроводная сеть
                                 Status:          Media disconnected
                           [02]: Realtek PCIe GbE Family Controller
                                 Connection Name: Ethernet
                                 DHCP Enabled:    Yes
                                 DHCP Server:     172.16.9.1
                                 IP address(es)
                                 [01]: 172.16.9.12
                                 [02]: fe80::7143:e2fe:cef5:8bc2
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

## Logs

<details>
<summary>Logs</summary>

```
Exception in thread "main" java.lang.NoClassDefFoundError: javax/xml/bind/annotation/XmlSchema
        at com.android.repository.api.SchemaModule$SchemaModuleVersion.<init>(SchemaModule.java:156)
        at com.android.repository.api.SchemaModule.<init>(SchemaModule.java:75)
        at com.android.sdklib.repository.AndroidSdkHandler.<clinit>(AndroidSdkHandler.java:81)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:73)
        at com.android.sdklib.tool.sdkmanager.SdkManagerCli.main(SdkManagerCli.java:48)
Caused by: java.lang.ClassNotFoundException: javax.xml.bind.annotation.XmlSchema
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 5 more
```

```
[√] Flutter (Channel stable, 2.0.4, on Microsoft Windows [Version 10.0.17763.1817], locale ru-RU)
    • Flutter version 2.0.4 at C:\flutter
    • Framework revision b1395592de (11 days ago), 2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\emakey\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.55.1)
    • VS Code at C:\Users\emakey\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 89.0.4389.114

! Doctor found issues in 1 category.
```

</details>

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/80273

The latest android studio can only run the cmdline tools version of the sdk manager. Running the platform tools version will lead to an unresolvable `java.lang.NoClassDefFoundError`

### CODE REVIEW

**Comment by jonahwilliams** [comment]
In order for this change to land, all of our infra recipes would need to be updated to download the cmdline-tools

File: packages/flutter_tools/lib/src/android/android_sdk.dart [hunk 1]
```diff
@@ -262,7 +268,7 @@ class AndroidSdk {
     return null;
   }
 
-  String? getCmdlineToolsPath(String binaryName) {
+  String? getCmdlineToolsPath(String binaryName, {bool skipTools = false}) {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/android/android_sdk.dart:271]: Is there a more specific name than `skipTools` for this flag? Maybe `skipOldTools` (supposing that I'm reading the comments in the function correctly)?

Comment (zanderso) [reply packages/flutter_tools/lib/src/android/android_sdk.dart:271]: Also does the `skipTools = true` case have test coverage?

**Comment by jonahwilliams** [comment]
Made the doctor check fatal, now waiting on an update to LUCI

File: packages/flutter_tools/lib/src/android/android_sdk.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/android/android_sdk.dart]: done!

**Review by zanderso — APPROVED** [review]

**Pull request merged**